File: result/metagpt/claude/humaneval/8.c
#include <stdio.h>

void sum_product(int numbers[], int size, int *sum, int *product) {
    *sum = 0;
    *product = 1;
    for (int i = 0; i < size; i++) {
        *sum += numbers[i];
        *product *= numbers[i];
    }
}

File: result/metagpt/gpt4o/humaneval/8.rs
fn sum_product(numbers: &[i32]) -> (i32, i32) {
    let mut sum = 0;
    let mut product = 1;

    for &number in numbers.iter() {
        sum += number;
        product *= number;
    }

    (sum, product)
}

fn main() {
    let numbers = vec![1, 2, 3, 4, 5];
    let (sum, product) = sum_product(&numbers);
    println!("Sum: {}, Product: {}", sum, product);
}